mod-pubsub-client/src/main/java/org/folio/rest/client/WebClientProvider.java

```diff
@@ -0,0 +1,20 @@
+package org.folio.rest.client;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+
+public class WebClientProvider {
+  private static final Map<Vertx, WebClient> webClients = new HashMap<>();
+
+  private WebClientProvider() {
+  }
+
+  public static synchronized WebClient getWebClient(Vertx vertx) {
+    Objects.requireNonNull(vertx);
+    return webClients.computeIfAbsent(vertx, WebClient::create);
+  }
```

Comment:
```suggestion
  public static synchronized WebClient getWebClient(@NonNull Vertx vertx) {
    return webClients.computeIfAbsent(vertx, WebClient::create);
  }
```
I saw you add lombok dependency, so why didn't you use it ? --> import lombok.NonNull

Reply:
yes, make sense. It was changed before adding Lombok, seems I forgot this part.

Reply:
done